lib/view_component/errors.rb

```diff
@@ -1,16 +1,213 @@
 module ViewComponent
-  class ViewContextCalledBeforeRenderError < StandardError; end
+  class BaseError < StandardError
+    def initialize
+      super(self.class::MESSAGE)
+    end
+  end
+
+  class DuplicateSlotContentError < StandardError
+    MESSAGE =
+      "It looks like a block was provided after calling `with_content` on COMPONENT, " \
+      "which means that ViewComponent doesn't know which content to use.\n\n" \
+      "To fix this issue, use either `with_content` or a block."
```

Comment:
Would these look better with squiggly heredocs?

```suggestion
    MESSAGE = <<~MESSAGE
      It looks like a block was provided after calling `with_content` on COMPONENT,
      which means that ViewComponent doesn't know which content to use.

      To fix this issue, use either `with_content` or a block.
    MESSAGE
```

Reply:
I'm not sure these lines are equivalent, though? I'm trying to have the first sentence of the output be a single line of text.

Reply:
Confirmed in the console: `=> "It looks like a block was provided after calling `with_content` on COMPONENT,\n`